### DISCUSSION TITLE
QuickScore: For better match in search and autocomplete

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5723] [Ideas]
I am asking can we use QucikScore to improve the search result in Tiddlywiki?
It seems it can also be used for autocomplete!

QuickScore improves on the original Quicksilver algorithm by tuning the scoring for long strings, such as webpage titles or URLs, so that the order of the search results makes more sense. It's used by the QuicKey extension for Chrome to enable users to easily find an open tab via search.

QuickScore is fast, dependency-free, and is just 2KB when minified and gzipped.

https://github.com/fwextensions/quick-score

**Comment by pmario** [comment]
The minified version is about 9kByte. gzipped is "used" if a server uses "server side compression" to send the library over the wire. ... With TW we need to "embed" a full or minified version into the core or a plugin

**Comment by Jermolene** [comment]
Thanks @kookma, that one is new to me.

I've investigated a number of more ambitious full text search JS libraries such as http://elasticlunr.com/ and https://lunrjs.com/ but it may be that quick-score is easier to integrate with our existing code.

In any case, my preferred approach to this would be to add enough hooks to the core code so that it's easy to install a custom search handler within a plugin.

**Reply by linonetwo** [reply]
And hope the hook can be async, so we can run search lib in a separate worker thread, or maybe even other nodejs thread when using tw in electron.

**Comment by pmario** [comment]
@Jermolene ... There is a plugin that uses https://lunrjs.com/ ... I think it's a nice one. ... The only problem with that plugin is, that it didn't activate "fuzzy-search" by default. ... see: https://hoelz.ro/files/fts.html

**Reply by pmario** [reply]
The other point I don't like, is the typescript to javascript overhead it introduces. I've had a look at the code and think about 40% of the size could be removed.
